docs/tutorial-aks-demo.md

```diff
@@ -0,0 +1,386 @@
+# Secure AKS Secrets with Key Vault
+
+## Introduction
+
+A Secret in Kubernetes is meant to save sensitive and secure data like passwords, certificates, and keys. These Secret objects aren't secret or secure! They're encoded using Base64 and saved into *etcd*. Also, by default, any pod can access to them. As a solution, we can encrypt data at rest as explained in [Kubernetes documentation](https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data). In this case, it's recommended to use a Key Management System (KMS) to securely save the encryption keys.
+
+KMS systems themselves have the feature to save the sensitive data like keys, secrets, files, and certificates. So, why not simply using it instead! Even more than just encryption, they offer powerful features like key rotation and expiration date.
+
+Now, to access KMS, a password is needed and should be saved in a Secret object in *etcd*! We returned back to the same problem. Fortunately, cloud providers have a solution. Instead of using a password to access KMS, the managed Kubernetes cluster will be granted access to KMS. In Azure, for example, AKS can connect to Key Vault or SQL Database… using an Identity. This connection could be done using the open-source [project Pod Identity](https://github.com/Azure/aad-pod-identity). It uses Azure AD to create an Identity and assign the roles and resources.
```

Comment:
As this tutorial explains Secrets in KeyVault would it make sense to focus more on KeyVault vs. KMS?

Do you think a diagram could help explaining your scenario?

Reply:
Also replaced that with:


Another solution would be using Azure Key Vault. It can securely encrypt sensitive data like keys, secrets, files, and certificates. Even more than just encryption, they offer powerful features like key rotation, expiration date and access policies.

Now, to access Key Vault, a password is needed. Thus it will be non-securely saved in a Secret object in *etcd*! We returned back to the first problem. Fortunately in Azure, AKS can connect to Key Vault or SQL Database… using an Identity. This connection could be done using the open-source [project Pod Identity](https://github.com/Azure/aad-pod-identity). It uses Azure AD to create an Identity and assign the roles and resources.

Reply:
I'll try to create a diagram

Reply:
Let me know what do you think about this: https://youtu.be/kr5rl3H3Mwo